pf_driver/src/pf/pf_interface.cpp

```diff
@@ -149,10 +188,16 @@ void PFInterface::stop_transmission()
 {
   if (state_ != PFState::RUNNING)
     return;
-  pipeline_->terminate();
-  pipeline_.reset();
-  protocol_interface_->stop_scanoutput(info_.handle);
-  protocol_interface_->release_handle(info_.handle);
+
+  // Unfortunately this node as it is now cannot join the reader & writer threads properly if the scanner is not
+  // accessible. we will just let the process die and let the OS do the cleaning up.
```

Comment:
have you checked whether the clean termination methods of the reader/writer had any important side effects (such as communication of the termination to the scanner, or important log messages?)